ICAFIX/scripts/computeRecleanFeatures.m

```diff
@@ -0,0 +1,273 @@
+        
+        VOLUME = reshape(VOLUME, prod(size(VOLUME, [1 2 3])), size(VOLUME, 4));
+        volumeValid = range(VOLUME, 2) > 0;
+        Volume_data = VOLUME(volumeValid, :);
+        
+        [wmparc_data, WMPARC] = vol_reshape_and_mask(WMPARC, volumeValid);
+        wm_data = vol_reshape_and_mask(WM, volumeValid);
+        csf_data = vol_reshape_and_mask(CSF, volumeValid);
+        nongray_data = vol_reshape_and_mask(NONGREY, volumeValid);
+        dropout_data = vol_reshape_and_mask(DROPOUT, volumeValid);
+        sbref_data = vol_reshape_and_mask(SBREFOrig, volumeValid);
+        
+        wm_data(wm_data ~= 0) = 1;
+        csf_data(csf_data ~= 0) = 1;
+        noisy_data=(sum(Volume_data,2)./sbref_data)>prctile(sum(Volume_data,2)./sbref_data,87.5);
+        edge_data=wmparc_data==0;
+        Volume_data=Volume_data./std(Volume_data(:));
+        CIFTI.cdata=CIFTI.cdata./std(CIFTI.cdata(:));
+        
+        nongray_data(nongray_data ~= 0) =1;
+
+        NonGrey=nongray_data;
+        NonGreyPlusEdge=NonGrey+edge_data;
+        Grey=(NonGreyPlusEdge-1)*-1;
+        VolSmoothROIs_data=wmparc_data.*0;
+        VolSmoothROIs_data(NonGrey==1)=1;
+        VolSmoothROIs_data(Grey==1)=2;
+        VolSmoothROIs_data(edge_data==1)=3;
+        VOLSMOOTHROIS=WMPARC.*0;
+        VOLSMOOTHROIS(std(VOLUME,[],2)>0)=VolSmoothROIs_data;
+        VOLSMOOTHROIS=reshape(VOLSMOOTHROIS,size(SBREFOrig,1),size(SBREFOrig,2),size(SBREFOrig,3),size(SBREFOrig,4));
+        save_avw(VOLSMOOTHROIS,[SubjFolderlist '/MNINonLinear/ROIs/VolumeSmoothROIs.' Resolution '.nii.gz'],'f',[1.6 1.6 1.6 1]);
+        system(['fslcpgeom ' SubjFolderlist '/MNINonLinear/Results/' fMRIName '/' fMRIName '_SBRef.nii.gz ' SubjFolderlist '/MNINonLinear/ROIs/VolumeSmoothROIs.' Resolution '.nii.gz -d']);
+        
+        SmoothingSigma=FinalSpatialSmoothingFWHM/(2*sqrt(2*log(2)));
+        system(['wb_command -volume-label-import ' SubjFolderlist '/MNINonLinear/ROIs/VolumeSmoothROIs.' Resolution '.nii.gz "" ' SubjFolderlist '/MNINonLinear/ROIs/VolumeSmoothROIs.' Resolution '.nii.gz']);
+        system(['wb_command -volume-parcel-smoothing ' SubjFolderlist '/MNINonLinear/Results/' fMRIName '/' fMRIName '_hp' hp '.ica/filtered_func_data.ica/melodic_oIC.nii.gz ' SubjFolderlist '/MNINonLinear/ROIs/VolumeSmoothROIs.' Resolution '.nii.gz ' num2str(SmoothingSigma) ' ' SubjFolderlist '/MNINonLinear/Results/' fMRIName '/' fMRIName '_hp' hp '.ica/filtered_func_data.ica/melodic_oIC_FWHM' num2str(FinalSpatialSmoothingFWHM) '.nii.gz']);
+
+        VOLUMESMOOTH=read_avw([SubjFolderlist '/MNINonLinear/Results/' fMRIName '/' fMRIName '_hp' hp '.ica/filtered_func_data.ica/melodic_oIC_FWHM' num2str(FinalSpatialSmoothingFWHM) '.nii.gz']);
+        VOLUMESMOOTH=reshape(VOLUMESMOOTH,size(VOLUMESMOOTH,1)*size(VOLUMESMOOTH,2)*size(VOLUMESMOOTH,3),size(VOLUMESMOOTH,4));
+        VolumeSmooth_data=VOLUMESMOOTH(std(VOLUMESMOOTH,[],2)>0,:);
```

Comment:
```suggestion
        VolumeSmooth_data=VOLUMESMOOTH(volumeValid, :);
```
DANGER: this uses a different data source for the data masking, and is therefore not entirely guaranteed to give the same answer, however later it is used indexwise with NonGrey, therefore this line should be changed to use exactly the same mask.

Reply:
Actually, it should use the function to get rid of the reshape, too...